google-cloud-clients/google-cloud-contrib/google-cloud-nio/src/main/java/com/google/cloud/storage/contrib/nio/CloudStorageRetryHandler.java

```diff
@@ -34,26 +30,60 @@
   private long totalWaitTime; // in milliseconds
   private final int maxRetries;
   private final int maxReopens;
+  private final CloudStorageConfiguration config;
+
 
   /**
    * Create a CloudStorageRetryHandler with the maximum retries and reopens set to the same value.
    *
    * @param maxRetriesAndReopens value for both maxRetries and maxReopens
+   *
+   * @deprecated use CloudStorageRetryHandler(CloudStorageConfiguration) instead.
    */
   public CloudStorageRetryHandler(final int maxRetriesAndReopens) {
     this.maxRetries = maxRetriesAndReopens;
     this.maxReopens = maxRetriesAndReopens;
+    // we're just using the retry parameters from the config, so it's OK to have a default.
+    this.config = CloudStorageConfiguration.DEFAULT;
   }
 
-  /**
+    /**
    * Create a CloudStorageRetryHandler with the maximum retries and reopens set to different values.
    *
    * @param maxRetries maximum number of retries
    * @param maxReopens maximum number of reopens
+   *
+   * @deprecated use CloudStorageRetryHandler(CloudStorageConfiguration) instead.
    */
   public CloudStorageRetryHandler(final int maxRetries, final int maxReopens) {
     this.maxRetries = maxRetries;
     this.maxReopens = maxReopens;
+    // we're just using the retry parameters from the config, so it's OK to have a default.
+    this.config = CloudStorageConfiguration.DEFAULT;
+  }
+
+  /**
+   * Create a CloudStorageRetryHandler with the maximum retries and reopens set to the same value.
+   *
+   * @param config - configuration for reopens and retryable codes.
+   */
+  public CloudStorageRetryHandler(final CloudStorageConfiguration config) {
+    this.maxRetries = config.maxChannelReopens();
+    this.maxReopens = config.maxChannelReopens();
+    this.config = config;
+  }
+
+  /**
+   * Create a CloudStorageRetryHandler with the maximum retries and reopens set to different values.
+   *
+   * @param maxRetries maximum number of retries
+   * @param maxReopens maximum number of reopens
+   * @param config http codes we'll retry on, and exceptions we'll reopen on.
+   */
+  public CloudStorageRetryHandler(final int maxRetries, final int maxReopens, final CloudStorageConfiguration config) {
```

Comment:
Do we need this one if you can set the maxRetries and maxReopens from the config?

Reply:
We need this one because the config only holds maxReopens, not maxRetries. The maxReopen option there lets us override the setting in the config.

Reply:
Ah, I missed that. Does it make sense to add that configuration option to the CloudStorageConfiguration?

Reply:
That would make for a much bigger change than we were aiming for here. The reopen can be specified each time we open a connection (in the public API), so we'd have to createa a new config object just to carry this parameter to the retry handler.